File: Greed/src/spacealert/humanPlayerInterface/commands/actionCards/FlipCardOnAndroidActionBoardCommand.java
package spacealert.humanPlayerInterface.commands.actionCards;

import com.fasterxml.jackson.annotation.JsonProperty;
import spacealert.core.planningPhase.IPlanningPhaseExposedToDecisionMaker;
import spacealert.core.planningPhase.IPlayerExposedToDecisionMaker;

import java.util.UUID;

public class FlipCardOnAndroidActionBoardCommand implements IPlanningPhaseCommand {
    @JsonProperty
    private UUID cardId;

    public FlipCardOnAndroidActionBoardCommand(UUID cardId) {
        this.cardId = cardId;
    }

    @Override
    public void execute(IPlanningPhaseExposedToDecisionMaker planningPhase, IPlayerExposedToDecisionMaker player) {
        planningPhase.flipCardOnAndroidActionBoard(cardId);
    }
}

File: Greed/src/client/spacealert/src/models/commands/FlipCardOnAndroidActionBoardCommand.ts
import { IsUUID } from "class-validator"

export default class FlipCardOnAndroidActionBoardCommand {
  type = ".FlipCardOnAndroidActionBoardCommand"

  @IsUUID()
  cardId: string

  constructor(cardId: string) {
    this.cardId = cardId
  }
}